p2p/src/simulated/network.rs

```diff
@@ -571,15 +587,19 @@ impl<E: RNetwork + Spawner + Rng + Clock + GClock + Metrics, P: PublicKey> Netwo
             };
 
             // Check rate limit for this (sender, channel) pair
+            // Note: We use try_lock here since we're in a synchronous context.
+            // If the lock is held (rare), we skip the check and let the message through.
             if let Some(limiter) = self.rate_limiters.get(&(origin.clone(), channel)) {
-                if limiter.check_key(&recipient).is_err() {
-                    trace!(
-                        ?origin,
-                        ?recipient,
-                        reason = "rate limited",
-                        "dropping message"
-                    );
-                    continue;
+                if let Some(guard) = limiter.try_lock() {
```

Comment:
The lock will also only be held over a synchronous piece of code in `sender`? I think we can just acquire.